processor/stream/stream_processor_test.go

```diff
@@ -156,3 +158,34 @@ func TestIntegration(t *testing.T) {
 		})
 	}
 }
+
+func TestRateLimiting(t *testing.T) {
+	report := func(ctx context.Context, p publish.PendingReq) error {
+		for range p.Transformables {
+		}
+		return nil
+	}
+
+	b, err := loader.LoadDataAsBytes("../testdata/intake-v2/ratelimit.ndjson")
+	require.NoError(t, err)
+	for _, test := range []struct {
+		name string
+		lim  *rate.Limiter
+		hit  int
+	}{
+		{name: "NoLimiter"},
+		{name: "LimiterDenyAll", lim: rate.NewLimiter(rate.Limit(0), 2)},
+		{name: "LimiterAllowAll", lim: rate.NewLimiter(rate.Limit(40), 40*5)},
+		{name: "LimiterAllowWithWait", lim: rate.NewLimiter(rate.Limit(10), 10*2), hit: 10},
+		{name: "LimiterForbiddenHit", lim: rate.NewLimiter(rate.Limit(7), 7*2), hit: 10},
```

Comment:
what's the difference between `LimiterForbiddenHit` and `LimiterForbidden`?

tiny naming nit pick: Deny vs. Forbidden. I thought this name was meant to mean that it's a test for _hit_ting the forbidden limit. Maybe we should go with `Deny` here too to keep it clearly separate from "Method Forbidden".

"Hit" here means that some part of the limit has already been used as far as i understand. Maybe simply "limitUsed" or similar would make that clearer?